test/e2e/metrics_test.go

```diff
@@ -156,11 +159,43 @@ func (c *MetricsTestConfig) validate(token string) {
 	require.Contains(c.t, string(output), "200 OK", "Metrics endpoint did not return 200 OK")
 }
 
-// cleanup created resources
+// cleanup removes the created resources. Uses a context with timeout to prevent hangs.
 func (c *MetricsTestConfig) cleanup() {
 	c.t.Log("Cleaning up resources")
-	_ = exec.Command(c.client, "delete", "clusterrolebinding", c.clusterBinding, "--ignore-not-found=true").Run()
-	_ = exec.Command(c.client, "delete", "pod", c.curlPodName, "-n", c.namespace, "--ignore-not-found=true").Run()
+	_ = exec.Command(c.client, "delete", "clusterrolebinding", c.clusterBinding, "--ignore-not-found=true", "--force").Run()
+	_ = exec.Command(c.client, "delete", "pod", c.curlPodName, "-n", c.namespace, "--ignore-not-found=true", "--force").Run()
```

Comment:
Hi @bentito 

After we add the --force in your 10 runner tests what is happening. 
Do we still facing the issue and logging ( it was not possible delete the pod or clusterbinding ) or now the issue is no longer faced?

Reply:
Anyway, I would be OK with it for now, if works does not impact us /lgtm
Just for we keep in mind in the future.